packages/api/tests/util/index.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  AtpAgentFetchHandlerResponse,
+  ComAtprotoLabelDefs,
+  AppBskyFeedDefs,
+  AppBskyActorDefs,
+  AppBskyFeedPost,
+  AppBskyEmbedRecord,
+  AppBskyGraphDefs,
+} from '../../src'
+
+export async function fetchHandler(
+  httpUri: string,
+  httpMethod: string,
+  httpHeaders: Record<string, string>,
+  httpReqBody: unknown,
+): Promise<AtpAgentFetchHandlerResponse> {
+  // The duplex field is now required for streaming bodies, but not yet reflected
+  // anywhere in docs or types. See whatwg/fetch#1438, nodejs/node#46221.
+  const reqInit: RequestInit & { duplex: string } = {
+    method: httpMethod,
+    headers: httpHeaders,
+    body: httpReqBody
+      ? new TextEncoder().encode(JSON.stringify(httpReqBody))
+      : undefined,
+    duplex: 'half',
+  }
+  const res = await fetch(httpUri, reqInit)
+  const resBody = await res.arrayBuffer()
+  return {
+    status: res.status,
+    headers: Object.fromEntries(res.headers.entries()),
+    body: resBody ? JSON.parse(new TextDecoder().decode(resBody)) : undefined,
+  }
+}
+
+export const mock = {
+  post({
+    text,
+    reply,
+    embed,
+  }: {
+    text: string
+    reply?: AppBskyFeedPost.ReplyRef
+    embed?: AppBskyFeedPost.Record['embed']
+  }): AppBskyFeedPost.Record {
+    return {
+      $type: 'app.bsky.feed.post',
+      text,
+      reply,
+      embed,
+      langs: ['en'],
+      createdAt: new Date().toISOString(),
+    }
+  },
+
+  postView({
+    record,
+    author,
+    embed,
+    replyCount,
+    repostCount,
+    likeCount,
+    viewer,
+    labels,
+  }: {
+    record: AppBskyFeedPost.Record
+    author: AppBskyActorDefs.ProfileViewBasic
+    embed?: AppBskyFeedDefs.PostView['embed']
+    replyCount?: number
+    repostCount?: number
+    likeCount?: number
+    viewer?: AppBskyFeedDefs.ViewerState
+    labels?: ComAtprotoLabelDefs.Label[]
+  }): AppBskyFeedDefs.PostView {
+    return {
+      uri: `at://${author.did}/app.bsky.post/fake`,
```

Comment:
I noticed we use the real collection name `app.bsky.feed.post` above in `post()`, but not here.  Just flagging in case you think it's important for those to match for any reason.

Reply:
Whoops yep, that's a mistake